lib/couchdb.js

```diff
@@ -491,13 +491,18 @@ CouchDB.prototype.exists = function(model, id, options, cb) {
 CouchDB.prototype.find =
   CouchDB.prototype.findById = function(model, id, options, cb) {
     debug('CouchDB.prototype.find %j %j %j', model, id, options);
+    cb = cb || createPromiseCallback();
     const self = this;
     const mo = self.selectModel(model);
     mo.db.get(id, function(err, doc) {
       if (err && err.statusCode === 404) return cb(null, []);
       if (err) return cb(err);
+      // Check if the fetched record belong to the query model.
+      const modelView = mo.modelView;
+      if (doc[modelView] != model) return cb(new Error('Record not found for model %s', model));
```

Comment:
Let's make this error message more descriptive please.

1. Include the `id` in the error message
2. Explain that the id exists but holds a different model (e.g. "wanted model Product but found Category")


Also Error constructor does not support format strings like `%s`, you have to either build the string manually (e.g. using template string literals) or use `util.format`.

I feel this change is somewhat unrelated to adding promise support, it would be best to move it to a standalone commit (or a pull request).

Reply:
@bajtos Sure this PR will promisify the api only. I will create a new one for the error check.